gobblin-data-management/src/main/java/org/apache/gobblin/data/management/source/LoopingDatasetFinderSource.java

```diff
@@ -125,13 +145,24 @@ public WorkUnitStream getWorkunitStream(SourceState state) {
 
     private Iterator<PartitionableDataset.DatasetPartition> currentPartitionIterator;
     private int generatedWorkUnits = 0;
+    private String globalWatermarkDatasetUrn;
```

Comment:
It seems that this flag is a proxy for "should we use dataset state store". Why not use a `useDatasetStateStore` flag instead? The `globalWatermarkDatasetUrn` in general need not be modified by the user.